lib/puppet-lint/plugins/check_unsafe_interpolations.rb

```diff
@@ -0,0 +1,28 @@
+PuppetLint.new_check(:check_unsafe_interpolations) do
+  COMMANDS = Array['command', 'onlyif', 'unless']
+  def check
+    # Gather any exec commands' resources into an array
+    exec_resources = resource_indexes.map do |resource|
+      resource if resource[:type].value == 'exec' && !(COMMANDS & resource[:param_tokens].map(&:value)).empty?
+    end
+
+    # Iterate over each command found in any exec
+    exec_resources.each do |command_resources|
+      # Filter each resource in command and check for unsaafe interpolations
+      command_resources[:tokens].select do |token|
+        # Check if command contains an input variable
+        if token.type == :VARIABLE
```

Comment:
I wonder if we should add the check for a newline back in here? Otherwise we would keep iterating through all of the tokens available.

I think previously it was a while loop.. there may be a more elegant way to handle this though.

Reply:
i don't think this is needed as the `exec_resources` array will only contain tokens found in an exec command